tests/jupyterhub_api/test_stop_session.py

```diff
@@ -0,0 +1,150 @@
+"""
+
+Test to stop the session
+
+To run the test use the following command:
+python3 test_stop_session.py --port 443 --users user1 --path /srv/jupyterhub
+
+For multiple users
+
+python3 test_stop_session.py --port 443 --users user0 user1 user2 --path /srv/jupyterhub
+
+"""
+import json
+import requests
+import subprocess
+import os
+from SessionUtils import CreateSession
+import time
+import argparse
+import sys
+sys.path.append("..")
+import time
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+    parser.add_argument( "--users", nargs='+', dest="users",
+                        required  = True,
+                        help='list of users')
+    parser.add_argument("--path", dest="path",
+                        required=True,
+                        help = 'Path where config file exists')
+    args = parser.parse_args()
+    return args
+
+class StopSession:
+    def __init__(self, port, users, path):
+        self.port = port
+        self.users = users
+        self.path = path
+        self.main_url = "https://localhost:" + str(self.port) + "/hub/api/"
+        self.ref_test_name ="Stop_Session"
+        self.exit = 0
+        self.token = ""
+        self.session = CreateSession(port, users, path)
+        self.ref_timestamp = int(time.time())
+        self.logger_folder = os.path.join(os.getcwd(), LOG_FOLDER)
+        self.log = Logger(os.path.join(self.logger_folder,
+                                       self.ref_test_name + "_" + time.strftime("%Y-%m-%d_%H:%M:%S") + LOG_EXTENSION))
+        self.log_params()
+
+    def log_params(self):
+        self.log.write("parameters", "Test name: " + self.ref_test_name)
+        self.log.write("parameters", "Test time: " + str(self.ref_timestamp))
+        self.log.write("parameters", "Logger folder: " + self.logger_folder)
+
+
+
+    def create_session(self):
```

Comment:
As before: I would simply try to stop a session given the username.
If it exists and is stopped --> all good
If it does not exist or something failed while stopping --> test failed